camel-core/src/main/java/org/apache/camel/builder/ExpressionBuilder.java

```diff
@@ -1895,13 +1920,25 @@ public Object evaluate(Exchange exchange) {
                     throw new IllegalArgumentException("Command not supported for dateExpression: " + command);
                 }
 
-                SimpleDateFormat df = new SimpleDateFormat(pattern);
-                return df.format(date);
+                // Apply offsets
+                long dateAsLong = date.getTime();
+                for (long offset : offsets) {
+                    dateAsLong += offset;
+                }
+                date = new Date(dateAsLong);
+
+                if (null != pattern && !pattern.isEmpty()) {
+                    SimpleDateFormat df = new SimpleDateFormat(pattern);
+                    df.setTimeZone(TimeZone.getTimeZone(timezone));
```

Comment:
If the given timezone is invalid TimeZone will use GMT. I don't think we should add layers here either. The documentation is pretty clear that the user is dealing with a plain SimpleDateFormat here.